src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,47 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T, W> {
+    private T first;
+    private W second;
 
+    private Pair(T first, W second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public W getSecond() {
+        return second;
+    }
+
+    public static <T, W> Pair of(T first, W second) {
+        return new Pair<>(first, second);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || !obj.getClass().equals(Pair.class)) {
+            return false;
+        }
+        Pair pair = (Pair) obj;
```

Comment:
we should cast to our(parametrize) class. You can just ignore warning from Idea in this case